include/mbedtls/error.h

```diff
@@ -48,38 +48,37 @@
  * Low-level module errors (0x0002-0x007E, 0x0003-0x007F)
  *
  * Module    Nr  Codes assigned
- * BIGNUM     7  0x0002-0x0010
+ * BIGNUM     8  0x0002-0x0010
  * GCM        2  0x0012-0x0014
  * BLOWFISH   2  0x0016-0x0018
  * THREADING  3  0x001A-0x001E
  * AES        2  0x0020-0x0022
  * CAMELLIA   2  0x0024-0x0026
  * XTEA       1  0x0028-0x0028
  * BASE64     2  0x002A-0x002C
- * OID        1  0x002E-0x002E   0x000B-0x000B
+ * OID        2  0x002E-0x002E   0x000B-0x000B
  * PADLOCK    1  0x0030-0x0030
  * DES        1  0x0032-0x0032
- * CTR_DBRG   4  0x0034-0x003A
- * ENTROPY    3  0x003C-0x0040   0x003D-0x003F
+ * CTR_DRBG   4  0x0034-0x003A
+ * ENTROPY    5  0x003C-0x0040   0x003D-0x003F
  * NET       11  0x0042-0x0052   0x0043-0x0045
  * ASN1       7  0x0060-0x006C
- * PBKDF2     1  0x007C-0x007C
  * HMAC_DRBG  4                  0x0003-0x0009
  * CCM        2                  0x000D-0x000F
  *
  * High-level module nr (3 bits - 0x0...-0x7...)
  * Name      ID  Nr of Errors
  * PEM        1  9
  * PKCS12     1  4 (Started from top)
- * X509       2  20
+ * X509       3  20
```

Comment:
Ouch, this reveals a mistake that has been made when allocating `MBEDTLS_ERR_X509_FATAL_ERROR`: it should clearly be `0x2A00` not `0x3000`. This is very unfortunate is has already been published and fixing it is an ABI change now (I added it to our list).

I think the script should be changed to still output `2` as the ID of X509; I realise this means adding a hack to the script to get around a mistake in `x509.h` but I think that's the best way to go for now. In the future (when the x509 value is fixed) it could be good for the script to warn about this.